### PR TITLE
Validate or reject a form

### PR DESCRIPTION
**Pull request by sifelix**
image

### CODE REVIEW

File: packages/interfaces/src/kobo/PatchValidationStatusDto.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { IsEnum, IsIn, IsString } from 'class-validator';
+
+import { ValidationStatusValue } from './ValidationStatusDto';
+import { DisasterType, DROUGHT, FLOOD, INCIDENT } from './constants';
+
+export class PatchValidationStatusDto {
+  @IsIn([FLOOD, DROUGHT, INCIDENT])
+  readonly disasterType!: DisasterType;
```
Comment (p-chavoshi) [inline packages/interfaces/src/kobo/PatchValidationStatusDto.ts:8]: seems redundant?

File: apps/backend/src/modules/kobo/kobo.controller.ts [hunk 2]
```diff
@@ -62,4 +64,18 @@ export class KoboController {
       throw new HttpException('Not Found', HttpStatus.NOT_FOUND);
     }
   }
+
+  @Patch('form/validationStatus')
+  async patchValidationStatus(
+    @Body() body: PatchValidationStatusDto,
+  ): Promise<ValidationStatusDto> {
+    // TODO add province guard for PCDM
```
Comment (p-chavoshi) [inline apps/backend/src/modules/kobo/kobo.controller.ts:72]: so we send out a PatchDto and receive back a ValidationDto ?

File: apps/frontend/components/pages/Form/FormValidationContainer.tsx [hunk 3]
```diff
@@ -20,6 +26,35 @@ export const FormValidationContainer = (): JSX.Element => {
       ) : (
         <FormValidation validationForm={form} />
       )}
+      <Box>
+        <Box>{JSON.stringify(form?._validation_status)}</Box>
```
Comment (p-chavoshi) [inline apps/frontend/components/pages/Form/FormValidationContainer.tsx:30]: at this point form should always be defined, is this "?" for typescript?

File: apps/frontend/services/api/kobo/usePatchValidationStatus.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+) => {
+  const { trigger, isMutating } = useSWRMutation(
+    [ApiRoutes.form, formDisasterType, formId],
+    async (
+      [relativePath, disasterType, id],
+      { arg }: { arg: ValidationStatusValue },
+    ) => {
+      const url = path.join(relativePath, 'validationStatus');
+      const { data: updatedValidationStatus } =
+        await apiClient.patch<ValidationStatusDto>(url, {
```
Comment (p-chavoshi) [inline apps/frontend/services/api/kobo/usePatchValidationStatus.ts:24]: should it be < PatchValidationStatusDto >?

File: apps/frontend/components/pages/Form/FormValidationContainer.tsx (hunk 3 above)
Comment (sifelix) [reply apps/frontend/components/pages/Form/FormValidationContainer.tsx:30]: It is outside of the ternary here

**Review by p-chavoshi — APPROVED** [review]

**Pull request merged**